pr_agent/algo/utils.py

```diff
@@ -211,3 +211,27 @@ def load_large_diff(file, new_file_content_str: str, original_file_content_str:
         except Exception:
             pass
     return patch
+
+
+def update_settings_from_args(args):
+    if args and len(args) >= 1:
+        for arg in args:
+            try:
+                arg = arg.strip('-').strip()
+                vals = arg.replace('=', '.').replace('=', '.').split('.')
+                d = settings
+                for i, v in enumerate(vals[:-1]):
+                    if i == len(vals) - 2:
+                        if v in d:
+                            if type(d[v]) == bool:
+                                d[v] = vals[-1].lower() in ("yes", "true", "t", "1")
+                            else:
+                                d[v] = type(d[v])(vals[-1])
+                            logging.info(f'Updated setting {vals[:-1]} to: "{vals[-1]}"')
+                            break
+                        else:
+                            logging.error(f'Invalid setting {vals[:-1]}')
+                    else:
+                        d = d[v]
+            except Exception as e:
+                logging.error(f'Failed to parse argument {arg}: {e}')
```

Comment:
**Suggestion:** The `update_settings_from_args` function is not robust enough. It assumes that the configuration settings are always nested two levels deep. This may not be the case if the configuration structure changes in the future. It's better to make this function more generic to handle any level of nesting.
```suggestion
def update_settings_from_args(args):
    if args and len(args) >= 1:
        for arg in args:
            try:
                arg = arg.strip('-').strip()
                vals = arg.replace('=', '.').replace('=', '.').split('.')
                d = settings
                for i, v in enumerate(vals[:-1]):
                    if v in d:
                        d = d[v]
                    else:
                        logging.error(f'Invalid setting {vals[:-1]}')
                        break
                else:
                    if type(d) == bool:
                        d = vals[-1].lower() in ("yes", "true", "t", "1")
                    else:
                        d = type(d)(vals[-1])
                    logging.info(f'Updated setting {vals[:-1]} to: "{vals[-1]}"')
            except Exception as e:
                logging.error(f'Failed to parse argument {arg}: {e}')
```